pkg/controller/submariner/submariner_controller.go

```diff
@@ -244,13 +239,13 @@ func (r *ReconcileSubmariner) reconcileServiceDiscovery(submariner *submarinerv1
 				return err
 			}
 			if result == controllerutil.OperationResultCreated {
-				reqLogger.Info("Created Service Discovery CR", "Namespace", sd.Namespace, "Name", sd.Name)
+				reqLogger.Info("Created Service Discovery CR", "Namespace", submariner.Namespace, "Name", serviceDiscoveryCrName)
```

Comment:
Why did you change these from `sd.Namespace` and `sd.Name`? That's better so at least this code isn't affected should these change. `sd` is available here - please change it back.

Reply:
It was giving me an error like sd can be nil. So I had to change.

Reply:
Then you can just initialize `sd` on L217, ie move L219 to L217.